doc/scapy/layers/automotive.rst

```diff
@@ -447,6 +450,64 @@ Sending a CRO message::
 Since sr1 calls the answers function, our payload of the DTO objects gets interpreted with the
 command of our CRO object.
 
+
+Universal calibration and measurement protocol (XCP)
+====================================================
+
+XCP is the successor of CCP. It is usable with several protocols. Scapy includes CAN, UDP and TCP.
+XCP has two types of message types: Command Transfer Object (CTO) and Data Transmission Object (DTO).
+CTOs send to an ECU are requests (commands) and the ECU has to reply with a positive response or an error.
+Additionally, the ECU can send a CTO to inform the master about an asynchronous event (EV) or request a service execution (SERV).
+DTOs sent by the ECU are called DAQ (Data AcQuisition) and include measured values.
+DTOs received by the ECU are used for a periodic stimulation and are called STIM (Stimulation).
+
+
+Creating a CTO message::
+
+    CTORequest() / Connect()
+    CTORequest() / GetDaqResolutionInfo()
+    CTORequest() / GetSeed(mode=0x01, resource=0x00)
+
+To send the message over CAN a header has to be added
+
+    pkt = XCPOnCAN(identifier=0x700) / CTORequest() / Connect()
+    sock = CANSocket(iface=can.interface.Bus(bustype='socketcan', channel='vcan0'))
+    sock.send(pkt)
+
+If we are interested in the response of an ECU, we need to set the basecls parameter of the
+CANSocket to XCPonCAN and we need to use sr1:
+Sending a CTO message::
+
+    sock = CANSocket(iface=can.interface.Bus(bustype='socketcan', channel='vcan0'), basecls=XCPonCAN)
```

Comment:
This syntax is outdated. 
It would be `CANSocket(bustype='socketcan', channel='vcan0', basecls=XCPonCAN)` now